src/appmixer/servicenow/lib.js

```diff
@@ -0,0 +1,275 @@
+'use strict';
+const crypto = require('crypto');
+const pathModule = require('path');
+
+const IGNORE_PROPERTIES_INSPECTOR = ['sys_id'];
+
+function getBasicAuth(username, password) {
+    return Buffer.from(username + ':' + password).toString('base64');
+}
+
+function getCacheKey(obj) {
+    const str = JSON.stringify(obj);
+    return crypto
+        .createHash('sha256')
+        .update(str)
+        .digest('hex');
+}
+
+async function callEndpointCached(context, options) {
+
+    let lock;
+    try {
+        lock = await context.lock(context.componentId);
```

Comment:
Shouldn't the lock name be the same for all `CreateRecord` components with the same options? This will create a different lock for each component. The optimization in reducing API calls will be lost.